### PR TITLE
soft deprecate funs()

### PR DESCRIPTION
**Pull request by romainfrancois**
Still need to: 

 - actually soft deprecate `funs()`
 - should we keep the tests in `test-funs.R` 

I replaced tests and various mentions of `funs` in the documentation by using `list` (sometimes named).

### CODE REVIEW

**Comment by hadley** [comment]
I think for this release we should probably do a soft-deprecation (i.e. no warning), and then warn in the next release (i.e. file an issue to hard deprecate)

**Comment by romainfrancois** [comment]
I think I need help (probably from @lionel-) on this one: 

```r
test_that("can supply funs()", {
  skip("need help for this one")
  expect_identical(select_if(df, funs(is_integerish(.)), funs(toupper(.))), set_names(df[c("x", "z")], c("X", "Z")))
  expect_identical(rename_if(df, funs(is_integerish(.)), funs(toupper(.))), set_names(df, c("X", "y", "Z")))
})
```

**Comment by hadley** [comment]
Yeah, I'd recommend either `fun`, `~ fun(.x)`, or lists of either form. (Then mention `rlang::as_function()` in a parenthetical.

File: tests/testthat/test-underscore.R [hunk 1]
```diff
@@ -133,23 +133,6 @@ test_that("filter_ works", {
   )
 })
 
-test_that("funs_ works", {
```
Comment (krlmlr) [inline tests/testthat/test-underscore.R]: Note to self: Moved to `test-funs.R`.

**Pull request merged**